lib/Common/Exceptions/Throw.cpp

```diff
@@ -125,13 +125,22 @@ namespace Js {
         }
         throw OutOfMemoryException();
     }
-    void Throw::CheckAndThrowOutOfMemory(BOOLEAN status)
+
+#if !defined(_M_IX86) && defined(_WIN32)
+    void Throw::XDataRegistrationError(HRESULT hr, ULONG_PTR funcStart)
     {
-        if (!status)
+        ULONG_PTR            imageBase = 0;
+        RUNTIME_FUNCTION  *runtimeFunction = RtlLookupFunctionEntry(funcStart, &imageBase, nullptr);
+        Unused(runtimeFunction);
+
+        if (JsUtil::ExternalApi::GetCurrentThreadContextId())
         {
-            OutOfMemory();
+            XDataRegistration_unrecoverable_error(hr);
```

Comment:
put the runtimeFunction in the exception context to preserve it and avoid compiler optimizing it away.